libraries/resource_providers/api200/logical_interconnect_provider.rb

```diff
@@ -38,7 +38,7 @@ def interconnect_handler(present_block, absent_block)
       # Recursive helper method to set hash values
       def recursive_set(actual, updates)
         updates.each_pair do |k, v|
-          if v.respond_to?(:each_pair)
+          if v.respond_to?(:each_pair) && actual[k].respond_to?(:each_pair)
```

Comment:
I've added ```actual[k].respond_to?(:each_pair)``` in the if clause because if to be passed a hash with attributes and these attributes is not present in the actual hash, an undefined method occurs: ```undefined method [] = 'for nil: NilClass```.